app/services/spree/feeds/base.rb

```diff
@@ -13,21 +13,36 @@ def generate
       end
 
       # This is used in all feed types to reduce the volume of SQL queries
-      def variants_includes
+      def variants_includes_base
         [
           :images,
           {
             product: [
               { product_properties: :property },
-              { master: [:images, :variant_image_images] },
+              { master: master_variant_includes },
               { taxons: :taxonomy },
             ],
           },
           { option_values: :option_type },
-          :variant_image_images,
           :default_price,
         ]
+      end
 
+      def master_variant_includes
+        [:images].tap do |mv_includes|
+          if Spree::Variant.respond_to?(:variant_image_images)
+            mv_includes << :variant_image_images
+          end
+        end
+      end
+
+      def variants_includes
+        if Spree::Variant.respond_to?(:variant_image_images)
+          base = variants_includes_base
+          base << :variant_image_images
```

Comment:
for these shovel operators, is there a chance we can add multiple duplicate `:variant_image_images` to this array?

Reply:
Maaaaaaybe? All of the code involved fits on one page right now, so someone would have to re-add those symbols to `variants_includes_base` while ignoring the method calls below.

Reply:
It could be rewritten as `base = base | [:variant_image_images]` which would ensure no duplication, but that syntax is just unusual enough to give me pause.

Reply:
agreed, that's a little tough to read.  haven't tested that theory on my end, but i've been burnt enough by shoveling items into an enumerable without checking if that item is already present i just wanted to throw it out there

Reply:
Agree 100%. If I wasn't building the entire struct from scratch right here, I would probably lean that way.